extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import { IPlatformService } from './platformService';
+import { BigDataClusterDeploymentProfile } from './bigDataClusterDeploymentProfile';
+
+interface BdcConfigListOutput {
+	stdout: string[];
+}
+
+export interface IAzdataService {
+	getDeploymentProfiles(): Thenable<BigDataClusterDeploymentProfile[]>;
+}
+
+export class AzdataService implements IAzdataService {
+	constructor(private platformService: IPlatformService) {
+	}
+
+	getDeploymentProfiles(): Thenable<BigDataClusterDeploymentProfile[]> {
+		return this.ensureWorkingDirectoryExists().then(() => {
+			return this.getDeploymentProfileNames();
+		}).then((names: string[]) => {
+			return Promise.all(names.map(name => { return this.getDeploymentProfileInfo(name); }));
+		});
+	}
+
+	private async getDeploymentProfileNames(): Promise<string[]> {
+		const env: NodeJS.ProcessEnv = {};
+		// azdata requires this environment variables to be set
+		env['ACCEPT_EULA'] = 'yes';
+		const cmd = 'azdata bdc config list -o json';
```

Comment:
Why is this only included in this call? Is it just because it happens to be first? If that's the case I'd strongly recommend you instead split it into its own separate function like `acceptAzDataEula` that just runs azdata without any args to do this work and then call that function as needed (when the wizard starts up probably). Although see my other comment below about that being necessary after the first run...

Reply:
ideally you should need to do this after first use, azdata will save this information. but the cost of checking whether this is first use is more expensive than just setting an environment variable, we need to do a file read. and that is not a officially documented way. so I'd prefer to keep it this way.